prime-router/docs/proposals/0018-universal-search.md

```diff
@@ -0,0 +1,91 @@
+    - I want to sort reports sent to me by how many items they contain.
+- I want to gain insight into the data I receive by querying the metadata associated with items. See 
+[All facilities & providers](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?node-id=995%3A13474&scaling=min-zoom&page-id=496%3A6448&starting-point-node-id=995%3A13227&show-proto-sidebar=1).
+    - I want a list of all performing facilities sending data to me.
+    - I want a list of all ordering providers sending data to me.
+    - I want a list of all submitters sending data to me.
+    - I want to know the last time a particular performing facility sent data to me.
+    - I want to know the last time a particular ordering provider sent data to me.
+    - I want to know the last time a particular submitter sent data to me.
+- Given some report item data, I want to find other data associated with the item. See 
+[Carroll Schultz](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?node-id=1081%3A15935&scaling=min-zoom&page-id=496%3A6448&starting-point-node-id=995%3A13227&show-proto-sidebar=1).
+    - I want to see when an ordering provider first reported to us, the receiver.
+    - I want the average number of tests per report sent to us for all reports including a particular ordering provider.
+    - I want the total number of items associated with a particular ordering provider.
+    - I want the contact information for a particular ordering provider.
+    - I want the CLIA associated with a particular ordering provider.
+
+**1.4** As a member of the engagement team (RS Admin), I want visibility into the data that flows through RS so I can 
+better troubleshoot issues. See **Message Tracker** hidden feature on RS website. See 
+[Engagement Engineer Document](https://docs.google.com/document/d/18Sk0NxBdn4K_tuMwBbhBdvfDtPjJ3wnEklg6i7taoAE/edit).
+- I want to find the metadata associated with a particular report item given a non-unique piece of metadata, like a 
+messageID in the case of a COVID message.
+- I want to search report items based on the date they were created or the testing lab they are associated with.
+- I want to search reports based on a date range.
+- I want to search items of reports based on a date range.
+- I want to view the metadata associated with a particular report.
+- I want to view the metadata associated with a particular item of a report.
+- Given a unique item identifier, I want all the metadata associated with the item, including the report(s) it belongs to. See **Message Tracker**.
+
+**1.5** As a member of the engagement team (RS Admin), I want to see failed actions and warnings in a specific 
+time range. [Reference Document](https://docs.google.com/document/d/18Sk0NxBdn4K_tuMwBbhBdvfDtPjJ3wnEklg6i7taoAE/edit).
+- I want to get all actions for a sender within a particular time frame. Currently this query times out consistently.
+
+# Design
+
+TODO
```

Comment:
Might be good to change this to "Coming soon!" or "The design will be documented at a later time." or something like that to make it more clear that this is intentionally incomplete and not something that we meant to fill out and forgot. Not a big deal, totally optional!